Update the Cumulocity user documentation to include details on the new Certificate Authority feature, covering device registration and CA creation. (#2985)
Cumulocity-IoT/c8y-docs: content/device-integration/device-enrollment-bundle/device-enroll-and-re-enroll.md

---
weight: 20
title: Simple device enrollment
layout: redirect
---

> In this tutorial, you will learn how to create device certificate which is signed by tenant's Certificate Authority(CA) with {{< product-c8y-iot >}}.

Reviewer: ```suggestion
In this tutorial, you will learn how to create a device certificate which is signed by a tenant's Certificate Authority (CA) with {{< product-c8y-iot >}}.
```

### Prerequisites {#prerequisites}

In order to follow this tutorial, check the following prerequisites:

* You have a valid tenant, a user and a password in order to access {{< product-c8y-iot >}}.

> * Certificate Authority (CA) is already created for this tenant.

Reviewer: ```suggestion
* A Certificate Authority (CA) has already been created for this tenant.
```

> For creating a new device certificate device enroll API need to be called. These following things happen when we call this API:

Reviewer: ```suggestion
On creating a new device certificate the device enroll API is called. This triggers the following actions:
```

> * Device enroll API to be used by a device to get a fresh new certificate.

Reviewer 2: It is also possible to register a device using EST via UI. You know I am referring to the "Create device certificate during device registration" option at UI. We need to describe the registration via UI as well.

Author: Done

Author: Done

> * Device enroll API to be used by a device to get a fresh new certificate. The device has to authenticate itself using its tenant, identifier and security token as the BasicAuth realm, user and password. These tenant, identifier and security token must be shared with Cumulocity using a '/devicecontrol/bulkNewDeviceRequests` onboarding endpoint to set the security token for a device.

Reviewer: Please rephrase. This is not understandable.

> * If a CA is not present then it will throw error.

Reviewer: ```suggestion
* If no CA is available an error will be returned.
```

> * If request does not contain valid [CertificateSigningRequest](https://en.wikipedia.org/wiki/Certificate_signing_request)

Reviewer: ```suggestion
* If the request does not contain a valid [CertificateSigningRequest](https://en.wikipedia.org/wiki/Certificate_signing_request)
```

Reviewer: It seems that something is missing here.

> This is example of REST request:

Reviewer: ```suggestion
This is an example of a REST request:
```
